src/anf-preview/azext_anf_preview/_params.py

```diff
@@ -0,0 +1,26 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+
+def load_arguments(self, _):
+    with self.argument_context('anf') as c:
+        c.argument('resource_group', options_list=['--resource-group', '-g'], required=True, help='The name of the resource group')
```

Comment:
You dont need this. The CLI has a default argument context for `resource_group`

Reply:
Yes I'd seen this and fully expected to be able to use either --resource-group or -g. However what I found was that without these lines in the _param file that specifying the command with -g resulted in 'argument --resource-group is required'.

Reply:
I see the problem, some of your resource_group parameter names are `resource_group`, while some are `resource_group_name`. They should all be `resource_group_name` to take advantage of the CLI's context. Please change the names to all match.

Reply:
Yes, that does indeed make it available for the custom commands. However list, show and delete still require the full --resource-group and cannot use -g without this in the _params.

Reply:
Interesting, I believe for the most part, the swagger def name for resource group is `resourceGroupName`, as indicated: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L291
A team member of mine has raised the following PR to address this: https://github.com/Azure/azure-rest-api-specs/pull/5256

If changing the swagger and generating new sdks is impossible for whatever reason, you will also be able to use: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/parameters.py#L236-L241